crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -118,7 +118,12 @@ impl<'a> Resolver<'a> {
 
         for unused_var in unused_vars.iter() {
             if self.interner.definition_name(unused_var.id) != ERROR_IDENT {
-                self.push_err(ResolverError::UnusedVariable { ident: *unused_var });
+                // Check whether the unused var is a global constant
+                let global_def_name = self.interner.definition_name(unused_var.id);
+                let global_ident = Ident::from(global_def_name.to_owned());
+                if self.interner.get_global_const(&global_ident).is_none() {
```

Comment:
This is legacy from when I was inserting const stmts directly into the function ast, thus they would be caught in the scope of every function and unless they were used in every function would cause an error (just another reason why that strategy was not good). Good catch, I removed it now. 

However, unused globals are still allowed. I held off originally on adding this logic, but I can make it so that globals error out when unused at all as well to be consistent with normal variables.